src/util/sighash.rs

```diff
@@ -144,8 +144,13 @@ impl SchnorrSigHashType {
         }
     }
 
-    /// Create a [`SchnorrSigHashType`] from raw `u8`
-    pub fn from_u8(hash_ty: u8) -> Result<Self, Error> {
+    /// Creates a [`SchnorrSigHashType`] from a raw `u8`.
+    ///
+    /// # Errors
+    ///
+    /// If the `hash_ty` value is non-standard (we include a 'special' 0x00 value that means 'use
+    /// default sighash flag' i.e., SIGHASH_ALL).
+    pub fn from_standard(hash_ty: u8) -> Result<Self, Error> {
```

Comment:
There is no standardness for `SchnorrSignaturehashType`. It is either valid or invalid. Ecdsa one has a standardness rule. This is also reflected in the error variant names. Perhaps this should be renamed back to `from_u8`?

Reply:
I view it as standardness here being equivalent with consensus validity. :) Does `from_consensus` seem less confusing to you?